File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let map = {}

        for (let i = 0; i < nums.length; i++) {

            if (nums[i] in map) {

                return [map[nums[i]], i]
            } else {

                let remainder = target - nums[i];

                map[remainder] = i;
            }
        }
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        res = [1] * len(nums)

        i = 1

        while (i < len(nums)):

            res[i] = res[i-1] * nums[i-1]
            i += 1
        
        j = len(nums) - 1
        suffix = 1
        while (j >= 0):

            res[j] *= suffix
            suffix *= nums[j]
            j -= 1
        
        return res